bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/discovery/CaddxDiscoveryService.java

```diff
@@ -0,0 +1,218 @@
+     */
+    public void addCaddxBridge(CaddxProtocol protocol, String port, int baudrate) {
+        logger.trace("addCaddxBridge(): Adding new Caddx Bridge on {} {} to Smarthome inbox", port, baudrate);
+
+        String bridgeID = "";
+        boolean containsChar = port.contains("/");
+
+        if (containsChar) {
+            String[] parts = port.split("/");
+            String id = parts[parts.length - 1].toUpperCase();
+            bridgeID = id.replaceAll("\\W", "_");
+        } else {
+            String id = port.toUpperCase();
+            bridgeID = id.replaceAll("\\W", "_");
+        }
+
+        Map<String, Object> properties = new HashMap<>(0);
+        properties.put(CaddxBridgeConfiguration.PROTOCOL, protocol);
+        properties.put(CaddxBridgeConfiguration.SERIAL_PORT, port);
+        properties.put(CaddxBridgeConfiguration.BAUD, baudrate);
+
+        try {
+            ThingUID thingUID = new ThingUID(CaddxBindingConstants.CADDXBRIDGE_THING_TYPE, bridgeID);
+
+            thingDiscovered(DiscoveryResultBuilder.create(thingUID).withProperties(properties)
+                    .withLabel("Caddx Bridge - " + port).build());
+        } catch (Exception e) {
+            logger.warn("addBridge(): ", e);
+        }
+    }
+
+    /**
+     * Method to add a Thing to the Smarthome Inbox.
+     *
+     * @param bridge
+     * @param caddxThingType
+     * @param event
+     */
+    public void addThing(Bridge bridge, CaddxThingType caddxThingType, CaddxEvent event) {
+        // logger.trace("addThing(): Adding new Thing {} to the smarthome inbox", caddxThingType.getLabel());
+
+        ThingUID thingUID = null;
+        String thingID = "";
+        String thingLabel = "";
+        Map<String, Object> properties = null;
+
+        Integer partition = event.getPartition();
+        Integer zone = event.getZone();
+        Integer keypad = event.getKeypad();
+
+        switch (caddxThingType) {
+            case PANEL:
+                thingID = "panel";
+                thingLabel = "Panel";
+                thingUID = new ThingUID(CaddxBindingConstants.PANEL_THING_TYPE, bridge.getUID(), thingID);
+                break;
+            case PARTITION:
+                thingID = "partition" + String.valueOf(partition);
+                thingLabel = "Partition " + String.valueOf(partition);
+                thingUID = new ThingUID(CaddxBindingConstants.PARTITION_THING_TYPE, bridge.getUID(), thingID);
+
+                properties = new HashMap<>(0);
+                if (partition != null) {
+                    properties.put(CaddxPartitionConfiguration.PARTITION_NUMBER, partition);
+                }
+                break;
+            case ZONE:
+                thingID = "zone" + String.valueOf(zone);
+                thingLabel = "Zone " + String.valueOf(zone);
+                thingUID = new ThingUID(CaddxBindingConstants.ZONE_THING_TYPE, bridge.getUID(), thingID);
+
+                properties = new HashMap<>(0);
```

Comment:
Why are you setting the initial size to 0? If you want to avoid the performance hit of allocating more memory set them to the minimum you might need. Which in this case is 1.
```suggestion
                properties = new HashMap<>(1);
```

Reply:
Done